src/behaviors/focusZone.ts

```diff
@@ -343,9 +343,18 @@ export function focusZone(container: HTMLElement, settings?: FocusZoneSettings):
   const activeDescendantControl = settings?.activeDescendantControl
   const activeDescendantCallback = settings?.onActiveDescendantChanged
 
-  function updateTabIndex(from?: HTMLElement, to?: HTMLElement) {
+  let activeDescendantSuspended = Boolean(activeDescendantControl)
+  let currentFocusedElement = activeDescendantControl ? undefined : focusableElements[0]
+
+  function updateFocusedElement(to?: HTMLElement) {
+    const from = currentFocusedElement
+    currentFocusedElement = to
+
     if (!activeDescendantControl) {
-      from?.setAttribute('tabindex', '-1')
+      if (from && from !== to && savedTabIndex.has(from)) {
+        from?.setAttribute('tabindex', '-1')
```

Comment:
Is optional chaining necessary here, given the `if (from)` check on the preceding line?

Reply:
Good catch! We need a lint rule for that